tools/data-handler/src/containers/project/resource-collector.ts

```diff
@@ -352,6 +361,44 @@ export class ResourceCollector {
     return readJsonFile(join(found.path, basename(found.name)));
   }
 
+  /**
+   *
+   * @param type
+   * @param name
+   * @param from
+   * @todo: this could use resource() or vice versa
+   * @returns
+   */
+  public async resourceObject(
+    type: string,
+    name: string,
+    from: ResourcesFrom,
+  ): Promise<object> {
```

Comment:
object type is bad, since `() => {}`, `{key: "value"}` and `[1]` are all objects. It could return resourceObject or map type to correct class.

Reply:
Yes. It could either be `CardTypeResource | ...` or their base class.